### PR TITLE
Tenant resend emails

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This adds a Tenant endpoint to resend verification emails by making a forwarded request to Quarterdeck.

Fixes SC-20578

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

- Endpoint should not be authenticated since this is called before the auth boundary
- The request should accept an email address, however we need to be careful about the response (e.g. if the email is not found we don't want to return 404 because it leaks information about user email addresses)

### Definition of Done

- [x] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [x] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [x] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/api/v1/client_test.go [hunk 1]
```diff
@@ -277,6 +277,33 @@ func TestVerifyEmail(t *testing.T) {
...
+
+	// Create a test server
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, http.MethodPost, r.Method)
+		require.Equal(t, "/v1/resend", r.URL.Path)
+
+		w.Header().Add("Content-Type", "application/json; charset=utf-8")
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode(fixture)
+	}))
```
Comment (bbengfort) [inline pkg/tenant/api/v1/client_test.go:291]: Might be good to port the serve fixture helper from Quarterdeck to Tenant to reduce code in the future.

File: pkg/tenant/auth.go [hunk 2]
```diff
@@ -461,3 +462,56 @@ func (s *Server) VerifyEmail(c *gin.Context) {
...
+	if err = c.BindJSON(&params); err != nil {
+		sentry.Warn(c).Err(err).Msg("could not parse resend email request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(responses.ErrTryResendAgain))
+		return
+	}
+
+	// Email is always required for this endpoint
+	req := &qd.ResendRequest{}
+	req.Email = strings.TrimSpace(params.Email)
+	if req.Email == "" {
```
Comment (bbengfort) [inline pkg/tenant/auth.go:492]:
```suggestion
	if req.Email == "" || len(req.Email) > 254 {
```

File: pkg/tenant/auth.go [hunk 3]
```diff
@@ -461,3 +462,56 @@ func (s *Server) VerifyEmail(c *gin.Context) {
...
+			sentry.Warn(c).Str("org_id", params.OrgID).Err(err).Msg("could not parse orgID from the request")
+			c.JSON(http.StatusBadRequest, api.ErrorResponse(responses.ErrBadResendRequest))
+			return
+		}
+	}
+
+	// Make the resend request to Quarterdeck
+	// Note: We are relying on Quarterdeck to adhere to best security practices and
+	// only return an error if the request is not parseable. Otherwise, it should
+	// return 204.
```
Comment (bbengfort) [inline pkg/tenant/auth.go:509]: Thank you for adding this comment!

File: pkg/tenant/auth_test.go [hunk 4]
```diff
@@ -440,3 +440,57 @@ func (s *tenantTestSuite) TestVerifyEmail() {
 	err = s.client.VerifyEmail(ctx, req)
 	s.requireError(err, http.StatusBadRequest, "invalid token")
 }
+
+func (s *tenantTestSuite) TestResendEmail() {
```
Comment (bbengfort) [inline pkg/tenant/auth_test.go:444]: I have to say the tests with the subtests are MUCH more readable.

**Review by bbengfort — APPROVED** [review]

File: pkg/tenant/auth_test.go (hunk 4 above)
Comment (pdeziel) [reply pkg/tenant/auth_test.go:444]: Agreed, they are a lot easier to review

**Pull request merged**